### PR TITLE
Fix out-of-office for rc version >4.0

### LINKED ISSUE #19: out-of-office does not work anymore since 4.0
Hello Graywolf336.

Since 4.x the out-of-office app is not working anymore. After enabling it there is no notification if someone sends a message to the out-of-office enabled user. There is nothing in the logs. 

Setup: Rocketchat 4.1.2, Kubernetes, 4 Rocketchat Nodejs Pods. 

Is there something I can do, to help finding the underlaying issue here?

cheers, t.

### PR DESCRIPTION
**Pull request by ceriath**
Fixes #19 and #18
Generally bumping versions and removing the username functions, since they are (official) deprecated since RocketChat 2.0.0 and apparently don't work anymore.

I also made a release until this is merged and published: https://github.com/ceriath/RocketChatApps/releases/tag/1.2.1

### CODE REVIEW

File: out-of-office/app.json [hunk 1]
```diff
@@ -3,13 +3,16 @@
     "name": "Out of Office Responder",
     "nameSlug": "out-of-office",
     "description": "Provides a command and auto responder for being out of office: /out-of-office",
-    "version": "1.2.0",
-    "requiredApiVersion": "^1.4.0",
+    "version": "1.3.0",
+    "requiredApiVersion": "^1.19.0",
```
Comment (graywolf336) [inline out-of-office/app.json:7]: Is this the lowest version supported? I’d like to support as many Rocket.Chat versions as possible. But if the bug doesn’t effect older versions, and this is merely a patch, then that’s okay

File: out-of-office/app.json [hunk 2]
```diff
@@ -3,13 +3,16 @@
     "name": "Out of Office Responder",
     "nameSlug": "out-of-office",
     "description": "Provides a command and auto responder for being out of office: /out-of-office",
-    "version": "1.2.0",
-    "requiredApiVersion": "^1.4.0",
+    "version": "1.3.0",
```
Comment (graywolf336) [inline out-of-office/app.json:6]: Since no new features are being added, let’s just do a patch increase if you don’t mind

**Comment by graywolf336** [comment]
Thank you for doing this!

File: out-of-office/app.json (hunk 1 above)
Comment (ceriath) [reply out-of-office/app.json]: So actually the old 1.4.0 API already supports the new functions, its just the new version not having the old functions. So i simply reverted that change.

File: out-of-office/app.json (hunk 2 above)
Comment (ceriath) [reply out-of-office/app.json]: Yes, i moved to 1.2.1

File: out-of-office/app.json [hunk 3]
```diff
@@ -3,13 +3,16 @@
...
         "support": "https://github.com/graywolf336/RocketChatApps/blob/master/out-of-office/README.md"
     },
     "classFile": "OutOfOfficeApp.ts",
-    "iconFile": "out-of-office.jpg"
-}
+    "iconFile": "out-of-office.jpg",
+    "implements": [
+        "IPostMessageSent"
+    ]
+}
```
Comment (graywolf336) [inline out-of-office/app.json:18]:
```suggestion
}

```

Comment (ceriath) [reply out-of-office/app.json]: done.

**Pull request merged**